src/domain/actuator.py

```diff
@@ -1,27 +1,36 @@
 """Actuator module"""
 
-import logging
 from abc import abstractmethod
+import datetime
 from enum import Enum
+import logging
+from queue import Empty, Queue
 
 logger = logging.getLogger(__name__)
 
-# List of Actuators
-actuators = []
-
 
 class Actuator:
     """Base class of an actuator."""
 
-    class ActuatorTypes(Enum):
-        Semaphore = "Semaphore"
-        Feeder = "Feeder"
+    actuators_pool = {}
 
-    def __init__(self, id, enabled=False):
-        self.type = None
-        self.id = id
+    def __init__(self, actuator_id, enabled=False):
+        self.cmd_queue = Queue()
+        self.actuator_id = actuator_id
+        self.last_update = None
         self.enabled = enabled
         self.stop_thread = False
+        Actuator.actuators_pool[self.actuator_id] = self
+
+    def send_command(self, msg: Enum):
+        self.cmd_queue.put(msg.value)
+
+    def get_command(self):
+        self.last_update = datetime.datetime.now()
+        try:
+            return self.cmd_queue.get(block=False)
+        except Empty:
+            return None
```

Comment:
should we log something here?

Reply:
I don't think so. The queue object raise an exception when it's empty but you ask for an element. I think the flow is correct and there is no need to log anything: an actuator asks for a command to execute, the queue is empty and it raises an exception, the exception is handled and the actuator receives a None, indicating there is no command to execute. Let me know what you think ;)